Remove PythonTeX Python environments and commands from LaTeX inspections

golmschenk opened the pull request:
#### Summary of additions and changes

* Excludes PythonTeX Python code environments and commands from LaTeX inspections (treating them as verbatim the same way Lua code is).
* PythonTeX Python code environments will require the `pythontex` package to be imported.

#### How to test this pull request

Content within the `pycode` environment or the `\py`, `\pyb`, `\pyc`, `\pys`, `\pyv` commands will not be inspected by LaTeX inspections.

#### Notes

The main purpose of this pull request is to remove the false-positive inspections from PythonTeX Python code. I used how Lua is currently handled as a guide. The basic idea there is that Lua code content is treated as verbatim.

Differing from the current Lua implementation, I did not add the Python language injection. I excluded this because the PythonTeX package can gobble indentation, and the Python code can be separated across segments of PythonTeX (e.g., a variable defined in one `pycode` environment can be used in another). The simple injection currently being used by the Lua implementation does not allow these and then still results in false-positive inspections (this time from the Python inspections). I am interested in working on this at some point, but this is a more complex task, and I'm not sure when/if I will attempt this. For reference, it appears this would require using `LanguageInjectionPerformer` and `LanguageInjectionContributor` (relevant issues related to injected language [indentation trimming](https://youtrack.jetbrains.com/issue/IDEA-214466/temporal-language-injection-should-support-multi-line-string-literals-with-margins-Scala-Kotlin) and [segment concatenation](https://youtrack.jetbrains.com/issue/KTIJ-855)). I haven't looked into these thoroughly yet, though. So, for now, the Python is just not inspected.

I didn't find much in the way of testing the Lua code environments. Please let me know if there are more tests that seem reasonable to add.

Please let me know what else might be needed to make this pull request acceptable. Thank you much!

PHPirates commented on src/nl/hannahsten/texifyidea/lang/commands/LatexListingCommand.kt:
```diff
@@ -19,6 +20,11 @@ enum class LatexListingCommand(
+    PYB("pyb", dependency = PYTHONTEX),
+    PYC("pyc", dependency = PYTHONTEX),
+    PYS("pys", dependency = PYTHONTEX),
+    PYV("pyv", dependency = PYTHONTEX),
```
If you add them here, you don't need to add them to LatexGenericRegularCommand

golmschenk replied:
I have removed the redundant definitions in `LatexGenericRegularCommand`. Just as a note, it appears `LUAEXEC` and `DIRECTLUA` are defined in both. Based on your comment, it sounded like you were suggesting `LatexGenericRegularCommand` is the duplication that should be removed. However, it looks like the `DIRECTLUA` and `LUAEXEC` that are in use are the ones from `LatexGenericRegularCommand`. Please let me know if you'd prefer that I switch around which ones are kept.

PHPirates replied:
Well spotted, thanks! I have removed those other duplicates as well

PHPirates reviewed (commented):
Awesome! Many thanks for adding this.

One small detail: because of a bug in a different plugin, we have to commit the generated files to git unfortunately, so I would appreciate if you also commit the regenerated lexer.
It's of course not how it should be, so I don't blame you for not noticing it, but you can read how to do it here: https://github.com/Hannah-Sten/TeXiFy-IDEA/wiki/Contributing-to-TeXiFy#editing-a-lexer

PHPirates commented:
Indeed, it very much looks like concatenation injection is nicely supported in the api. It's been a while since I used pythontex though, so I don't remember how hard it would be to figure out, but it would be really nice if we could get that to work.

If you need any help or maybe just something to get you started on this, please ask!

golmschenk commented:
I've removed the redundant command definitions (please see the note above in response to the original comment), and I've updated the generated lexer. Please let me know what else would be needed. Thank you much!

PHPirates reviewed (approved):
Thanks, looks good!

Also, has nothing to do with this PR but I noticed these commands did not appear in the autocompletion as they should, as the indexer does not pick them up. I'll try to fix that.
